ddbToEsDlqLambda/index.ts

```diff
@@ -0,0 +1,161 @@
+/* eslint-disable no-await-in-loop */
+import AWS from 'aws-sdk';
+import getComponentLogger from './loggerBuilder';
+
+const logger = getComponentLogger();
+const { QUEUE_URL, DDB_TO_ES_LAMBDA_NAME, MAX_NUMBER_OF_MESSAGES_TO_PROCESS } = process.env;
+const MESSAGE_BATCH_SIZE = 10;
+const MESSAGE_VISIBILITY_TIMEOUT = 30; // seconds
+
+const sqs = new AWS.SQS();
+const lambda = new AWS.Lambda();
+const dynamodbstreams = new AWS.DynamoDBStreams();
+
+const getMessages = async (queueUrl, batchSize) => {
+    const resp = await sqs
+        .receiveMessage({
+            QueueUrl: queueUrl,
+            MaxNumberOfMessages: batchSize,
+            VisibilityTimeout: MESSAGE_VISIBILITY_TIMEOUT,
+            WaitTimeSeconds: 0,
+        })
+        .promise()
+        .catch((e) => {
+            throw new Error(`Failed to receive messages: ${JSON.stringify(e)}`);
+        });
+
+    // when there are no messages, resp.Messages is undefined
+    const messages = resp.Messages === undefined ? [] : resp.Messages;
+    logger.debug('Received messages', JSON.stringify(messages));
+    return messages;
+};
+
+const deleteMessages = async (queueUrl, messagesToDelete) => {
+    const resp = await sqs
+        .deleteMessageBatch({
+            QueueUrl: queueUrl,
+            Entries: messagesToDelete,
+        })
+        .promise()
+        .catch((e) => {
+            throw new Error(`Failed to delete message: ${JSON.stringify(e)}`);
+        });
+
+    // resp.Successful contains successfully deleted messages.
+    if (resp.Successful) {
+        logger.debug('Deleted messages', JSON.stringify(resp.Successful));
+    }
+
+    // resp.Failed contains messages that were failed to delete
+    if (resp.Failed && resp.Failed.length) {
+        throw new Error(`Failed to delete messages: ${JSON.stringify(resp.Failed)}`);
+    }
+};
+
+/**
+ * This is asynchronous invocation, the successful invocation doesn't mean
```

Comment:
This is _an_ asynchronous invocation. A successful invocation doesn't mean a successful ....

Reply:
We actually need to make a synchronous invocation here in order to reliably remove messages from the queue once they have been processed.